funnel/assets/js/membership.js

```diff
@@ -56,25 +56,27 @@ const Membership = {
         };
       },
       methods: {
-        fetchForm(event, url, member = '') {
+        async fetchForm(event, url, member = '') {
           event.preventDefault();
           if (this.isUserProfileAdmin) {
             this.activeMember = member;
             const app = this;
-            $.ajax({
-              type: 'GET',
-              url,
-              timeout: window.Hasgeek.Config.ajaxTimeout,
-              dataType: 'json',
-              success(data) {
+            const response = await fetch(url, {
+              headers: {
+                Accept: 'application/json',
+                'X-Requested-With': 'XMLHttpRequest',
+              },
+            }).catch(Form.handleFetchNetworkError);
+            if (response && response.ok) {
+              const data = await response.json();
+              if (data) {
                 const vueFormHtml = data.form;
                 app.memberForm = vueFormHtml.replace(/\bscript\b/g, 'script2');
```

Comment:
I didn't see this earlier. Why is the word `script` being replaced with `script2`?

Reply:
This is for the script tag in the fetched html to work in vue framework
https://github.com/taoeffect/vue-script2

Reply:
We should be using the thorough regex used here: https://github.com/taoeffect/vue-script2#writing-script-instead-of-script2-using-script2ify. My version:

`/(?:</?script([\s>]))/gm`